src/sub_movie_generator.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+from typing import List, Optional
+from foxlator_lib.video import Video
+from foxlator_lib.stt.whisper_stt import WhisperSTT, AudioTextSegment
+
+from src.third_party.ffmpeg import check_ffmpeg_installed
+from src.third_party.imagemagick import check_imagemagick_installed
+
+
+class SubMovieGenerator:
+
+    def __init__(self, movie_path: Path, model_size: str, lang: Optional[str] = None):
+        if not check_ffmpeg_installed() or not check_imagemagick_installed():
+            exit()
```

Comment:
Having multiple exit points is generally not a great idea, I'd throw an exception here and catch it in the main function.

Reply:
Especially in constructor.